packages/flutter_tools/templates/create/ios.tmpl/Runner/AppDelegate.m

```diff
@@ -1,23 +1,16 @@
 #include "AppDelegate.h"
-{{#withPluginHook}}
-#include "{{pluginClass}}.h"
+#include "PluginRegistry.h"
 
 @implementation AppDelegate {
-  {{pluginClass}} *_{{pluginProjectName}};
+  PluginRegistry *plugins;
 }
-{{/withPluginHook}}
-{{^withPluginHook}}
-@implementation AppDelegate
-{{/withPluginHook}}
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
-    // Override point for customization after application launch.
-{{#withPluginHook}}
+  // Override point for customization after application launch.
   FlutterViewController *flutterController =
       (FlutterViewController *)self.window.rootViewController;
-  _{{pluginProjectName}} = [[{{pluginClass}} alloc] initWithFlutterView:flutterController];
-{{/withPluginHook}}
-    return YES;
+  plugins = [[PluginRegistry alloc] initWithController:flutterController];
```

Comment:
Similar to the Android concept discussed above, what if the PluginRegistry had a static +(instancetype)defaultRegistry getter and the FlutterViewControllers automatically registered themselves in their constructor. This would allow us to mess around with the auto-registration process without having to worry about changing all the client's AppDelegates every time the plugin registry changes.

AFAIK, there can only be only one AppDelegate per app, so having a global default plugin registry seems analogous to having the registry be owned by the FlutterApplication on Android.